src/op_handlers/shift.rs

```diff
@@ -0,0 +1,75 @@
+use crate::address_operands::{address_operands_read, address_operands_store};
+use crate::value::TaggedValue;
+use crate::{opcode::Opcode, state::VMState};
+use zkevm_opcode_defs::Opcode as Variant;
+use zkevm_opcode_defs::ShiftOpcode;
+
+fn _shl(vm: &mut VMState, opcode: &Opcode) {
+    let (src0_t, src1_t) = address_operands_read(vm, opcode);
+    let (src0, src1) = (src0_t.value, src1_t.value);
+    let shift = (src1.low_u32() % 256) as usize;
```

Comment:
Not saying this is _wrong_ per se, but I'm a bit confused.

The spec states that you take the lowest byte of the second input, here, you're taking the lowest 4 bytes, are we sure we aren't allowing something that shouldn't work based on the spec? We can also check how the fast vm is doing this to be completely sure. 

Plus, I think the usize cast is redundant, why can't it simply be a u32?

Reply:
I checked [fastVM](https://github.com/matter-labs/vm2/blob/9342db726462b76aa7e4ed246684b1316ea79c21/src/instruction_handlers/binop.rs#L106) and if we want something similar, we can use something like this instead:
```rust
let shift = src1.low_u32() as u8;
```
You are right that I get the last 4 bytes, but then I do the `mod 256` and I get the last bytes. I've done that so it's more aligned with the spec: 
```
- The result is computed as in1 << (in2 % 256).
```
But I don't know the performance or readability of the code in this case, should I change it to the above alternative?

Reply:
> But I don't know the performance or readability of the code in this case, should I change it to the above alternative?

As a rule of thumb, let's focus on simplicity+readability more than performance for now 🙂.

`- The result is computed as in1 << (in2 % 256)`
I'm a bit confused by the spec, it states you take the lowest byte as binary digits, so the max amount of shifts you can do
is 255, which is the range for numbers modulo 256. So the in2 % 256 looks a bit redundant. 
But for now, I think following the spec as is is the best thing to do, so let's only take the second argument modulo 256,
and ignore the low 4 bytes, as it seems redundant.
Let's double check this with more people though, pinging @jrchatruc @IAvecilla @gianbelinche.
Let's also have tests checking what happens when shifting with amounts greater than 256.

Reply:
I think both `src1.low_u32()` and `src1 % 256` would work.
I don't think the spec is redundant, since it asumes that in2 is a U256, so doing `in2 % 256` is actually the way of getting the last byte